Fix normals in export

Phanabani opened the pull request:
Nooga in the discord mentioned that the normals in the batch level exports were still broken. This pull request fixes normals in single and (almost all) batch exports. The only problem is that at the moment I can't get WriteObjMaterialwise to work. I'm using my winding-order helper function `shouldReverseWinding` just like everywhere else, but it's still exporting weirdly.

I started working on this before seeing your work on this same problem, and since I already know my version really well I just continued with the batch fix. Feel free to ignore this if you'd rather keep your changes! The main idea is the normals were not rotated by the object matrices in the batch export, so that's what I did here (starting at e0d8cbe) in addition to my earlier changes.

MilchRatchet commented:
Oh true, we did export the model space normals as if they were already world space. I will look at this code here. Since we need to resolve conflicts anyway, I guess it is best if I look into implementing the best of both implementations.

Phanabani commented:
Fixed the SeparateModelObjectByMaterial issue -- the indices were wrong. I also added a winding order check to the TerrainFragment writing in this method. I'm not sure what TerrainFragments are, but I assumed they would also need the winding order check.

Also I'm sorry if this pull request is an extra hassle! I figured I'd at least share it in case it is useful.
